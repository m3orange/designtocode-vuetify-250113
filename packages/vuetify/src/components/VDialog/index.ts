// Styles
import './VDialog.sass'

import VDialog from './VDialog'

export { VDialog }
export default VDialog
