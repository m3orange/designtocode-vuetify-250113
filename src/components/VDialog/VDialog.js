import VDialogContent from './VDialogContent'

let counter = 0

/* @vue/component */
export default {
  name: 'v-dialog',

  functional: true,

  $_wrapperFor: VDialogContent,

  render (h, context) {
    const slots = context.slots()
    const activator = slots.activator && slots.activator[0]

    context.data.ref = context.data.ref || '$_v-dialog-' + counter++

    const content = h(VDialogContent, context.data, slots.default)

    if (activator) {
      activator.data.on = activator.data.on || {}
      activator.data.on.click = e => {
        e.preventDefault()
        const dialog = context.parent.$refs[context.data.ref]
        if (!dialog.disabled) dialog.isActive = true
      }
    }

    return activator ? [activator, content] : content
  }
}
