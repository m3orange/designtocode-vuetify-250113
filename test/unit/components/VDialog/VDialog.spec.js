import VDialog from '@/components/VDialog'
import VDialogContent from '@/components/VDialog/VDialogContent'
import { test } from '@/test'

test('VDialog', ({ mount, compileToFunctions, functionalContext }) => {
  it('should render component and match snapshot', () => {
    const wrapper = mount(VDialog, functionalContext())

    expect(wrapper.html()).toMatchSnapshot()
    expect('Unable to locate target [data-app]').toHaveBeenTipped()
  })

  it('should render a disabled component and match snapshot', () => {
    const wrapper = mount(VDialog, functionalContext({
      props: {
        disabled: true
      }
    }))

    expect(wrapper.html()).toMatchSnapshot()
    expect('Unable to locate target [data-app]').toHaveBeenTipped()
  })

  it('should render a persistent component and match snapshot', () => {
    const wrapper = mount(VDialog, functionalContext({
      props: {
        persistent: true
      }
    }))

    expect(wrapper.html()).toMatchSnapshot()
    expect('Unable to locate target [data-app]').toHaveBeenTipped()
  })

  it('should render a fullscreen component and match snapshot', () => {
    const wrapper = mount(VDialog, functionalContext({
      props: {
        fullscreen: true
      }
    }))

    expect(wrapper.html()).toMatchSnapshot()
    expect('Unable to locate target [data-app]').toHaveBeenTipped()
  })

  it('should render a lazy component and match snapshot', () => {
    const wrapper = mount(VDialog, functionalContext({
      props: {
        lazy: true
      }
    }))

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render a scrollable component and match snapshot', () => {
    const wrapper = mount(VDialog, functionalContext({
      props: {
        scrollable: true
      }
    }))

    expect(wrapper.html()).toMatchSnapshot()
    expect('Unable to locate target [data-app]').toHaveBeenTipped()
  })

  it('should render component with custom origin and match snapshot', () => {
    const wrapper = mount(VDialog, functionalContext({
      props: {
        origin: 'top right'
      }
    }))

    expect(wrapper.html()).toMatchSnapshot()
    expect('Unable to locate target [data-app]').toHaveBeenTipped()
  })

  it('should render component with custom width (max-width) and match snapshot', () => {
    const wrapper = mount(VDialog, functionalContext({
      props: {
        maxWidth: 100
      }
    }))

    expect(wrapper.html()).toMatchSnapshot()
    expect('Unable to locate target [data-app]').toHaveBeenTipped()
  })

  it('should render component with custom width and match snapshot', () => {
    const wrapper = mount(VDialog, functionalContext({
      props: {
        width: '50%'
      }
    }))

    expect(wrapper.html()).toMatchSnapshot()
    expect('Unable to locate target [data-app]').toHaveBeenTipped()
  })

  it('should render component with custom transition and match snapshot', () => {
    const wrapper = mount(VDialog, functionalContext({
      props: {
        transition: 'fade-transition'
      }
    }))

    expect(wrapper.html()).toMatchSnapshot()
    expect('Unable to locate target [data-app]').toHaveBeenTipped()
  })

  it('should open dialog on activator click', async () => {
    const input = jest.fn()
    const component = {
      render (h) {
        return h('div', [h(
          VDialog,
          { on: { input } },
          [h('span', { slot: 'activator', staticClass: 'dialog__activator' }, 'activator')]
        )])
      }
    }
    const wrapper = mount(component)
    const content = wrapper.find(VDialogContent)[0]

    expect(content.vm.isActive).toBe(false)
    wrapper.find('.v-dialog__activator')[0].trigger('click')
    expect(content.vm.isActive).toBe(true)
    await wrapper.vm.$nextTick()
    expect(input).toBeCalledWith(true)

    expect('Unable to locate target [data-app]').toHaveBeenTipped()
  })

  it('should not open disabled dialog on activator click', async () => {
    const input = jest.fn()
    const component = {
      render (h) {
        return h('div', [h(
          VDialog,
          {
            on: { input },
            props: { disabled: true }
          },
          [h('span', { slot: 'activator', staticClass: 'dialog__activator' }, 'activator')]
        )])
      }
    }
    const wrapper = mount(component)
    const content = wrapper.find(VDialogContent)[0]

    expect(content.vm.isActive).toBe(false)
    wrapper.find('.v-dialog__activator')[0].trigger('click')
    expect(content.vm.isActive).toBe(false)
    await wrapper.vm.$nextTick()
    expect(input).not.toBeCalled()

    expect('Unable to locate target [data-app]').toHaveBeenTipped()
  })

  it('should not change state on v-model update', async () => {
    const input = jest.fn()
    const component = {
      props: { value: Boolean },
      render (h) {
        return h('div', [h(
          VDialog,
          {
            on: { input },
            props: { value: this.value }
          },
          [h('span', { slot: 'activator', staticClass: 'dialog__activator' }, 'activator')]
        )])
      }
    }
    const wrapper = mount(component)
    const content = wrapper.find(VDialogContent)[0]

    expect(content.vm.isActive).toBe(false)

    wrapper.setProps({
      value: true
    })
    await wrapper.vm.$nextTick()
    expect(content.vm.isActive).toBe(true)

    wrapper.setProps({
      value: false
    })
    await wrapper.vm.$nextTick()
    expect(content.vm.isActive).toBe(false)

    expect('Unable to locate target [data-app]').toHaveBeenTipped()
  })

  it('should emit keydown event', async () => {
    const keydown = jest.fn()
    mount({
      render: h => h(VDialog, {
        props: { value: true },
        on: { keydown }
      })
    })

    window.dispatchEvent(new Event('keydown'))
    expect(keydown).toBeCalled()

    expect('Unable to locate target [data-app]').toHaveBeenTipped()
  })
})
