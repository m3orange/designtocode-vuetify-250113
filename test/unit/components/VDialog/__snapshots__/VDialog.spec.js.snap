// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VDialog should render a disabled component and match snapshot 1`] = `

<div tabindex="-1"
     class="v-dialog__content"
     style="z-index: 0;"
>
  <div class="v-dialog"
       style="display: none;"
  >
  </div>
</div>

`;

exports[`VDialog should render a fullscreen component and match snapshot 1`] = `

<div tabindex="-1"
     class="v-dialog__content"
     style="z-index: 0;"
>
  <div class="v-dialog v-dialog--fullscreen"
       style="display: none;"
  >
  </div>
</div>

`;

exports[`VDialog should render a lazy component and match snapshot 1`] = `

<div tabindex="-1"
     class="v-dialog__content"
     style="z-index: 0;"
>
  <div class="v-dialog"
       style="display: none;"
  >
  </div>
</div>

`;

exports[`VDialog should render a persistent component and match snapshot 1`] = `

<div tabindex="-1"
     class="v-dialog__content"
     style="z-index: 0;"
>
  <div class="v-dialog v-dialog--persistent"
       style="display: none;"
  >
  </div>
</div>

`;

exports[`VDialog should render a scrollable component and match snapshot 1`] = `

<div tabindex="-1"
     class="v-dialog__content"
     style="z-index: 0;"
>
  <div class="v-dialog v-dialog--scrollable"
       style="display: none;"
  >
  </div>
</div>

`;

exports[`VDialog should render component and match snapshot 1`] = `

<div tabindex="-1"
     class="v-dialog__content"
     style="z-index: 0;"
>
  <div class="v-dialog"
       style="display: none;"
  >
  </div>
</div>

`;

exports[`VDialog should render component with custom origin and match snapshot 1`] = `

<div tabindex="-1"
     class="v-dialog__content"
     style="z-index: 0;"
>
  <div class="v-dialog"
       style="display: none;"
  >
  </div>
</div>

`;

exports[`VDialog should render component with custom transition and match snapshot 1`] = `

<div tabindex="-1"
     class="v-dialog__content"
     style="z-index: 0;"
>
  <div class="v-dialog"
       style="display: none;"
  >
  </div>
</div>

`;

exports[`VDialog should render component with custom width (max-width) and match snapshot 1`] = `

<div tabindex="-1"
     class="v-dialog__content"
     style="z-index: 0;"
>
  <div class="v-dialog"
       style="max-width: 100px; display: none;"
  >
  </div>
</div>

`;

exports[`VDialog should render component with custom width and match snapshot 1`] = `

<div tabindex="-1"
     class="v-dialog__content"
     style="z-index: 0;"
>
  <div class="v-dialog"
       style="width: 50%; display: none;"
  >
  </div>
</div>

`;
